Data/Unicode/Internal/Division.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE UnboxedTuples #-}
+-- |
+-- Module      : Data.Unicode.Internal.Division
+-- Copyright   : (c) 2020 Andrew Lelechenko
+--
+-- License     : BSD-3-Clause
+-- Maintainer  : harendra.kumar@gmail.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Fast division by known constants.
+
+module Data.Unicode.Internal.Division
+    ( quotRem21
+    , quotRem28
+    , isDivisibleBy28
+    ) where
+
+import Data.Bits (Bits(..), FiniteBits(..))
+import GHC.Exts (Word(..), timesWord2#)
+
+-- Input must be non-negative
+quotRem21 :: Int -> (Int, Int)
+quotRem21 n
+  | finiteBitSize (0 :: Word) /= 64
+  = n `quotRem` 21
+  | otherwise
+  = (fromIntegral q, fromIntegral (w - 21 * q))
+  where
+    w = fromIntegral n
+    high = highMul w 14054662151397753613 -- (2^68+17)/21
+    q = high `shiftR` 4
+{-# INLINE quotRem21 #-}
```

Comment:
We place a function definition before its first use in the file i.e. code is in bottom up order instead of top down order. The rationale is in this coding style document which captures the existing/intended coding style of all our projects here https://github.com/composewell/haskell-dev/blob/master/coding-style.rst . Feedback on coding style is also welcome, its been evolving.